tpl/images/images.go

```diff
@@ -117,3 +122,65 @@ func (ns *Namespace) Filter(args ...any) (images.ImageResource, error) {
 
 	return img.Filter(filtersv...)
 }
+
+var (
+	qrErrorCorrectionLevels = map[string]qr.Level{
+		"low":      qr.L,
+		"medium":   qr.M,
+		"quartile": qr.Q,
+		"high":     qr.H,
+	}
+	qrDefaultErrorCorrectLevel = "medium"
+)
+
+// QR encodes the given text to a QR code using the given error correction
+// level, returning an image resource.
+func (ns *Namespace) QR(args ...any) (images.ImageResource, error) {
+	if len(args) == 0 || len(args) > 2 {
+		return nil, errors.New("requires one or two arguments")
```

Comment:
This looks good in general. But re. 2 positional args vs a map is something I have thought about lately ... Don't make the change if you somehow disagree, but my take on this is:

* A map is a better choice for multiple args, and especially if it may become more in the future.
* The decoding becomes much simpler, you just create a default struct and use `mapstruct.WeakDecode`.
* It is in my head simpler to document, which becomes even more truthful if there comes more options in the future.